ecl/hql/hqlexpr.cpp

```diff
@@ -16188,6 +16618,292 @@ IHqlExpression * annotateIndexBlobs(IHqlExpression * expr)
+{
+    if (!comparison.nearest)
+        return;
+
+    if (nearest == comparison.nearest)
+        return;
+
+    unsigned unsignedDistance = comparison.distance == (unsigned)(-1) ? comparison.distance : comparison.distance + penalty;
+    if (unsignedDistance < distance)
+    {
+        distance = comparison.distance;
+        nearest.set(comparison.nearest);
+        penalty++;
+    }
+}
+
+void NearestSymbol::compare(IHqlExpression * comparison)
+{
+    if (!comparison || nearest == comparison)
+        return;
+
+    unsigned tempDistance = computeEditDistance(comparison->queryId());
+
+    if (tempDistance == 0 || tempDistance >= MAX_SYMBOL_DISTANCE)
+        return;
+
+    if (tempDistance < distance)
+    {
+        distance = tempDistance;
+        nearest.set(comparison);
+    }
+}
+
+unsigned NearestSymbol::computeEditDistance(const IIdAtom * comparison) const
+{
+    if (!comparison)
+        return -1;
+
+    const char * comparisonLowerStr = str(lower(comparison));
+    const char * searchNameLowerStr = str(lower(searchName));
+    if (strcmp(searchNameLowerStr, comparisonLowerStr) == 0)
+        return 0;
+
+    //The edit distance computation below is the full Damerau Levenshtein Distance with a weighted matrix.
+    //Therefore it is possible for it to return a distance of 0 yet the two strings not be identicle.
+    //For this reason +1 is added to differentiate if no weights are used, the restricted version is used, or the standard Levenshtein edit idstance is used.
+    switch(metric)
+    {
+    case optimalStringAlignment :
+        //call optimalStringAlignment
+    case damerauLevenshtein :
+        return damerauLevenshteinDistance(strlen(searchNameLowerStr), searchNameLowerStr, strlen(comparisonLowerStr), comparisonLowerStr, NULL, 0);
+    case weightedDamerauLevenshtein :
+        return damerauLevenshteinDistance(strlen(searchNameLowerStr), searchNameLowerStr, strlen(comparisonLowerStr), comparisonLowerStr, defaultConfusionMatrix, defaultConfusionMatrixLength) + 1;
+    };
+    return -1;
+}
+
+IHqlExpression * NearestSymbol::branchSearch(IHqlScope * scope, unsigned lookupFlags, HqlLookupContext & ctx, unsigned & penalty)
+{
+    NearestSymbol tempNearest(searchName);
+    scope->lookupNearestSymbol(searchName, lookupFlags, ctx, tempNearest);
+    compare(tempNearest);
+    return nearest;
+}
+
+IHqlExpression * NearestSymbol::branchSearch(IHqlSimpleScope * scope, unsigned & penalty)
+{
+    NearestSymbol tempNearest(searchName);
+    scope->lookupNearestSymbol(searchName, tempNearest);
+    compare(tempNearest, penalty);
+    return nearest;
+}
+
+void NearestSymbol::reset()
+{
+    nearest.clear();
+    distance = -1;
+}
+
+bool NearestSymbol::bestMatchFound() const
+{
+    if (distance > 1)
+        return false;
+
+    switch(metric)
+    {
+    case optimalStringAlignment :
+    case damerauLevenshtein :
+    {
+        if (distance == 1)
+            return true;
+    }
```

Comment:
missing break.  

style: simplest code would be

  case optimalStringAlignment :
  case damerauLevenshtein :
      return distance == 1;